drivers/sensor/tmag5273/tmag5273.c

```diff
@@ -0,0 +1,1118 @@
+			LOG_ERR("x-axis measurement deactivated");
+			return -ENOTSUP;
+		}
+		start_address = TMAG5273_REG_Z_MSB_RESULT;
+		end_address = TMAG5273_REG_Z_LSB_RESULT;
+		break;
+	case SENSOR_CHAN_MAGN_XYZ:
+		if (!drv_cfg->x_axis && !drv_cfg->y_axis && !drv_cfg->z_axis) {
+			LOG_ERR("xyz-axis measurement deactivated");
+			return -ENOTSUP;
+		}
+		start_address = TMAG5273_REG_X_MSB_RESULT;
+		end_address = TMAG5273_REG_Z_LSB_RESULT;
+		break;
+	case SENSOR_CHAN_DIE_TEMP:
+		if (!drv_cfg->temperature) {
+			LOG_ERR("temperature measurement deactivated");
+			return -ENOTSUP;
+		}
+		start_address = TMAG5273_REG_T_MSB_RESULT;
+		end_address = TMAG5273_REG_T_LSB_RESULT;
+		break;
+	case SENSOR_CHAN_ROTATION:
+		if (drv_cfg->angle_magnitude_axis == TMAG5273_ANGLE_CALC_NONE) {
+			LOG_ERR("axis measurement deactivated");
+			return -ENOTSUP;
+		}
+		start_address = TMAG5273_REG_ANGLE_MSB_RESULT;
+		end_address = TMAG5273_REG_ANGLE_LSB_RESULT;
+		break;
+	case TMAG5273_CHAN_MAGNITUDE:
+	case TMAG5273_CHAN_MAGNITUDE_MSB:
+		if (drv_cfg->angle_magnitude_axis == TMAG5273_ANGLE_CALC_NONE) {
+			LOG_ERR("axis measurement deactivated");
+			return -ENOTSUP;
+		}
+		start_address = end_address = TMAG5273_REG_MAGNITUDE_RESULT;
+		break;
+	case TMAG5273_CHAN_ANGLE_MAGNITUDE:
+		if (drv_cfg->angle_magnitude_axis == TMAG5273_ANGLE_CALC_NONE) {
+			LOG_ERR("axis measurement deactivated");
+			return -ENOTSUP;
+		}
+		start_address = TMAG5273_REG_ANGLE_MSB_RESULT;
+		end_address = TMAG5273_REG_MAGNITUDE_RESULT;
+		break;
+	case SENSOR_CHAN_ALL:
+		start_address = TMAG5273_REG_RESULT_BEGIN;
+		end_address = TMAG5273_REG_RESULT_END;
+		break;
+	default:
+		LOG_ERR("unknown sensor channel %d", chan);
+		return -EINVAL;
+	}
+
+	uint32_t nb_bytes = end_address - start_address + 1;
+	uint8_t buf_offset = start_address - TMAG5273_REG_RESULT_BEGIN;
+
+	while (nb_bytes) {
+		const uint8_t block_size = COND_CODE_1(
+			CONFIG_CRC, ((drv_cfg->crc_enabled && (nb_bytes > 4)) ? 4 : nb_bytes),
+			(nb_bytes));
+
+		retval = i2c_burst_read_dt(&drv_cfg->i2c, start_address,
+					   i2c_buffer + buf_offset, /* force same positioning */
+					   block_size);
+
+		if (retval < 0) {
+			LOG_ERR("could not read result data %d", retval);
+			return -EIO;
+		}
+
+#if CONFIG_CRC
+		/* check data validity, if activated */
+		if (drv_cfg->crc_enabled) {
+			const uint8_t crc = crc8_ccitt(0xFF, i2c_buffer + buf_offset, block_size);
+
+			if (i2c_buffer[buf_offset + block_size] != crc) {
+				LOG_ERR("invalid CRC value: 0x%X (expected: 0x%X)",
+					i2c_buffer[buf_offset + block_size], crc);
+				return -EIO;
+			}
+		}
+#endif
+
+		nb_bytes -= block_size;
+		buf_offset += block_size;
+	}
+
+	/* generic error check */
+	if (!drv_cfg->ignore_diag_fail) {
+		const uint8_t diag_status =
+			i2c_buffer[TMAG5273_REG_CONV_STATUS - TMAG5273_REG_RESULT_BEGIN] &
+			TMAG5273_DIAG_STATUS_MSK;
+
+		if (diag_status == TMAG5273_DIAG_FAIL) {
+			tmag5273_read_device_status(dev);
+			return -EIO;
```

Comment:
worth a LOG_WRN?

Reply:
Added a `LOG_ERR` to `tmag5273_read_device_status`, so that every branch generates an error/warning as needed.